### PR TITLE
Features/update annual income field #160776676

### PR DESCRIPTION
**Pull request by chrisdolendo**
https://www.pivotaltracker.com/story/show/160776676

### CODE REVIEW

File: spec/controllers/applications_controller_spec.rb [hunk 1]
```diff
@@ -24,7 +23,7 @@
     context 'with a non lease up application' do
       let(:expected_non_lease_up_app) { fixture('controllers/applications/non_lease_up_application_domain.json') }
 
-      it 'should return a domain application' do
+      it 'should return a domain application snapshot' do
```
Comment (akegan) [inline spec/controllers/applications_controller_spec.rb]: Hm, I think with a non-lease up application is should *not* return a snapshot, right?

**Review by akegan — DISMISSED** [review]
just one question. otherwise looks good

**Comment by akegan** [comment]
@chrisdolendo the ticket also needs review instructions

File: spec/controllers/applications_controller_spec.rb (hunk 1 above)
Comment (chrisdolendo) [reply spec/controllers/applications_controller_spec.rb]: Good catch, updated.

**Comment by chrisdolendo** [comment]
@akegan hola! tests pass and ready for your re-review

**Comment by akegan** [comment]
@chrisdolendo can you add more details to the review instructions around how this value should compare to the value in the snapshot view, and maybe a check on salesforce to make sure that it's saving to the correct field?

**Review by akegan — CHANGES_REQUESTED** [review]
Moving this comment to request changes just so it's clear that this PR is waiting on something:

@chrisdolendo can you add more details to the review instructions around how this value should compare to the value in the snapshot view, and maybe a check on salesforce to make sure that it's saving to the correct field?

File: app/javascript/components/supplemental_application/sections/ConfirmedHouseholdIncome.js [hunk 2]
```diff
@@ -44,13 +44,12 @@ const ConfirmedHouseholdIncome = ({ amis, amiCharts, formApi }) => {
           </FormGrid.Group>
         </FormGrid.Item>
         <FormGrid.Item>
-          <FormGrid.Group label='Household Annual Income'>
-            <Text field='annual_income'
+          <FormGrid.Group label='Confirmed Household Annual Income'>
```
Comment (akegan) [inline app/javascript/components/supplemental_application/sections/ConfirmedHouseholdIncome.js]: From the AC it looks like this should be updated to Confirmed Annual income so the field title doesn't wrap

File: app/javascript/components/mappers/domainToApi/application.js [hunk 3]
```diff
@@ -24,6 +24,7 @@ export const applicationFieldMapper = {
   has_developmental_disability: 'hasDevelopmentalDisability',
   answered_community_screening: 'answeredCommunityScreening',
   annual_income: 'annualIncome',
+  confirmed_annual_income: 'confirmedHouseholdAnnualIncome',
```
Comment (akegan) [inline app/javascript/components/mappers/domainToApi/application.js]: I'm also going to update this to be confirmed_household_annual_income. It's long but income fields are confusing enough that I think we need to be very consistent.

File: app/javascript/components/mappers/soqlToDomain/application.js [hunk 4]
```diff
@@ -29,6 +29,7 @@ export const mapApplication = (a) => {
     application_submission_type: a.Application_Submission_Type,
     application_submitted_date: a.Application_Submitted_Date,
     annual_income: a.Annual_Income,
+    confirmed_annual_income: a.Confirmed_Household_Annual_Income,
```
Comment (akegan) [inline app/javascript/components/mappers/soqlToDomain/application.js]: update here too

**Review by akegan — DISMISSED** [review]

**Comment by akegan** [comment]
I confirmed that the field is tied to the correct field on salesforce, and that the field label and helper text matches the AC:
image

@callaghanc I added some changes here so if you'd be up for a quick review on this that would be great

**Review by callaghanc — APPROVED** [review]

**Comment by akegan** [comment]
thank you for approving! did you also run through all of the review steps? would you be up for clicking "accept" on the ticket?

**Comment by callaghanc** [comment]
i did also do the Pivotal story review steps, so i'll accept the story in Pivotal

File: spec/fixtures/controllers/supplementals/index.json [hunk 5]
```diff
@@ -0,0 +1,394 @@
+{
```
Comment (akegan) [inline spec/fixtures/controllers/supplementals/index.json:1]: @chrisdolendo how did you generate this file? We need to update it but I'm not sure how you made it in the first place

**Pull request merged**